go/pools/resource_pool.go

```diff
@@ -325,6 +331,9 @@ func (rp *ResourcePool) SetCapacity(capacity int) error {
 func (rp *ResourcePool) recordWait(start time.Time) {
 	rp.waitCount.Add(1)
 	rp.waitTime.Add(time.Since(start))
+	if rp.name != "" {
+		rp.waitStats.Record(rp.name+"ResourceWaitTime", start)
```

Comment:
Doesn't the name get added automatically farther up the stack?

Reply:
Yes. As far as I know it is [only used](https://github.com/vitessio/vitess/blob/master/go/vt/vttablet/tabletserver/connpool/pool.go#L72) to publish stats.